plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -198,6 +198,39 @@ public function __construct( $args = array() ) {
 		// TODO: args to be implemented.
 		unset( $this->args['customer_note'], $this->args['name'] );
 
+		/**
```

Comment:
We shouldn't include filters in class constructors. It would be better to implement this in the build query function.

Reply:
@rrennick could you help me understand better why one would be preferable over the other? My thinking is: `build_query` is just called from the constructor, moving the filter in there doesn't really change anything control-flow-wise ...?

Reply:
@lsinger Good/consistent coding practices. I haven't done a count but I expect there are very few if any class constructors in WC that have hooks or filters in them.

Reply:
@rrennick just moving it into the `build_query` method doesn't feel very meaningful to me, though. If we want to improve the readability of the constructor, what would you think of going this route? Assuming we're in the constructor:

```php
$this->build_query();
if ( ! $this->maybe_override_query() ) {
  $this->run_query();
}
```

`maybe_override_query` would then contain the filter application and the following conditionals.

Reply:
@lsinger That sounds great.